src/databaseaccessor/connection.cpp

```diff
@@ -0,0 +1,35 @@
+#include "connection.h"
+#include <QDebug>
+
+Connection::Connection()
+{
+     mDb = QSqlDatabase::addDatabase(typeDatabase);
+     qDebug()<<"connection";
+}
+
+Connection::~Connection()
+{
+    mDb.close();
+    qDebug()<<"closed";
+}
+
+void Connection::openDatabase()
+{
+    mDb = QSqlDatabase::database();
+    mDb.setDatabaseName(mPath);
+    if(mDb.open())
+        qDebug()<<"opened";
+}
```

Comment:
We have to agree on strategy for cases when connection fails
You might raise custom exception (e.g. DBConnectionError) which will be caught in calling method